File: 300-longest-increasing-subsequence/longest-increasing-subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int mapper [nums.size()];
        fill_n(mapper, nums.size(), 1);
        for (int i = nums.size() - 1; i >=0 ; i--)
            for (int j = i; j < nums.size(); j++)
                if (nums[i] < nums[j]) mapper[i] = max(mapper[i], 1 + mapper[j]);
            
        return * max_element(mapper, mapper + (sizeof(mapper) / sizeof(mapper[0])));
    }
};

File: 300-longest-increasing-subsequence/longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        mapper = [1] * len(nums)
        for i in range(len(nums) - 1, -1, -1):
            for j in range(i, len(nums)):
                if nums[i] < nums[j]:
                    mapper[i] = max(mapper[i], 1 + mapper[j])
        return max(mapper)